src/tools/kata-ctl/src/exec.rs

```diff
@@ -0,0 +1,337 @@
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::{
+    io::{self, BufRead, BufReader, Read, Write},
+    os::unix::{
+        io::{AsRawFd, RawFd},
+        net::UnixStream,
+    },
+    path::PathBuf,
+    time::Duration,
+};
+
+use anyhow::anyhow;
+use glob::glob;
+use serde::{Deserialize, Serialize};
+use slog::debug;
+use vmm_sys_util::terminal::Terminal;
+
+const KATA_RUN_PATH: &str = "/run/kata";
+const KATA_HYBRID_VSOCK: &str = "kata.hvsock";
+const CONNECT_CMD: &str = "CONNECT";
+const OK_CMD: &str = "OK";
+
+// 5 seconds
+const KATA_AGENT_VSOCK_TIMEOUT: u64 = 5;
+
+// Convenience macro to obtain the scope logger
+#[macro_export]
+macro_rules! sl {
+    () => {
+        slog_scope::logger()
+    };
+}
+
+#[derive(Debug)]
+pub enum Error {
+    EpollWait(io::Error),
+    EpollCreate(io::Error),
+    EpollAdd(io::Error),
+    SocketWrite(io::Error),
+    StdioErr(io::Error),
+}
+
+type Result<T> = std::result::Result<T, Error>;
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Config {
+    pub server_address: String,
+    pub hybrid_vsock: bool,
+    pub hybrid_vsock_port: u64,
+    pub bundle_dir: String,
+    pub interactive: bool,
+}
+
+enum EpollDispatch {
+    Stdin,
+    ServerSock,
+}
+
+struct EpollContext {
+    epoll_raw_fd: RawFd,
+    stdin_index: u64,
+    dispatch_table: Vec<EpollDispatch>,
+    stdin_handle: io::Stdin,
+    debug_console_sock: Option<UnixStream>,
```

Comment:
I am not sure whether it is necessary to use Option<...> here, seems every time we use debug_console.sock, we just do `.as_mut.unwrap()` in process_handler.

Reply:
I will initialize the EpollContext.debug_console_sock, and the `.as_mut.unwrap()` is not in loop, It will do once when process_handler called.